service/resources/email.py

```diff
@@ -34,67 +47,81 @@ def send_email(data, resp):
-            "attachment": HelperService.get_attachments,
-            "tracking_settings": HelperService.get_email_trackings,
-            "custom_arg": HelperService.get_custom_args,
-            "section": HelperService.get_sections,
-            "header": HelperService.get_headers,
-            "category": HelperService.get_category
+            "attachments": HelperService.get_attachments,
+            "custom_args": HelperService.get_custom_args
+            # "section": HelperService.get_sections,
+            # "header": HelperService.get_headers,
+            # "category": HelperService.get_category
         }
 
         message.to = func_switcher.get("to")(data['to'], 'to')
-        if 'cc' in data.keys():
+        data_keys = data.keys()
+        if 'cc' in data_keys:
             message.cc = func_switcher.get("cc")(data['cc'], 'cc')
-        if 'bcc' in data.keys():
+        if 'bcc' in data_keys:
             message.bcc = func_switcher.get("bcc")(data['bcc'], 'bcc')
-        if 'content' in data.keys():
+        if 'template' in data_keys and not 'content' in data_keys:
+            data['content'] = generate_template_content(data['template'])
+            data_keys = data.keys()
+        if 'content' in data_keys:
             message.content = func_switcher.get("content")(data['content'])
-        if 'attachment' in data.keys():
-            message.attachment = func_switcher.get("attachment")(data['attachments'])
-        if 'tracking_settings' in data.keys():
-            message.tracking_settings = func_switcher.get("tracking_settings")(data['tracking_settings'])
-        if 'custom_arg' in data.keys():
-            message.custom_arg = func_switcher.get("custom_arg")(data['custom_args'])
-        if 'section' in data.keys():
-            message.section = func_switcher.get("section")(data['sections'])
-        if 'header' in data.keys():
-            message.header = func_switcher.get("header")(data['headers'])
-        if 'category' in data.keys():
-            message.category = func_switcher.get("category")(data['categories'])
-
-        #pylint: disable=broad-except
-        try:
-            #logging.warning(message.get())
-            sendgrid_api_key = os.environ.get('SENDGRID_API_KEY')
-
-            sendgrid_client = SendGridAPIClient(sendgrid_api_key)
-            response = sendgrid_client.send(message)
-            resp.body = response.body
-            resp.status = falcon.HTTP_200
-        except HTTPError as error:
-            print("error")
-            print(error.to_dict)
-        except Exception as error:
-            print("exception")
-            resp.body = json.dumps(str(error))
+        if 'attachments' in data_keys:
+            message.attachment = func_switcher.get("attachments")(data['attachments'])
+        if 'custom_args' in data_keys:
+            message.custom_arg = func_switcher.get("custom_args")(data['custom_args'])
+        # if 'section' in data_keys:
```

Comment:
To improve readability, can remove these if not used?

Reply:
removed commented code
https://github.com/SFDigitalServices/email-microservice-py/pull/5/commits/ac2fd5c36f5c4596779c0a288030eccaa8d42c97